pkg/filesystem/directory.go

```diff
@@ -79,6 +79,8 @@ type Directory interface {
 	Lstat(name string) (FileInfo, error)
 	// Mkdir is the equivalent of os.Mkdir().
 	Mkdir(name string, perm os.FileMode) error
+	// Mknod is the equivalent of unix.Mknod().
+	Mknod(name string, perm os.FileMode, dev int) error
```

Comment:
I think that Linux uses uint64 for the device number. This interface tries to be OS independent (because it can be implemented arbitrarily). Should we simply use uint64 here as well?

Reply:
When doing this I tried to match the underlying Mknod call which takes the device number as an int. would it be better pass a uint64 and cast it to an int within the directory.Mknod function, or to just leave it as it is?

Reply:
Hmmm... I just did a `git grep` on the go-sys package:

```
$ git grep 'func Mknodat' unix
unix/syscall_freebsd.go:func Mknodat(fd int, path string, mode uint32, dev uint64) (err error) {
unix/zsyscall_aix_ppc.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_aix_ppc64.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_dragonfly_amd64.go:func Mknodat(fd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_linux.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_netbsd_386.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_netbsd_amd64.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_netbsd_arm.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_netbsd_arm64.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_openbsd_386.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_openbsd_amd64.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_openbsd_arm.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_openbsd_arm64.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
unix/zsyscall_solaris_amd64.go:func Mknodat(dirfd int, path string, mode uint32, dev int) (err error) {
```
It looks like all OSes use `int`, except for FreeBSD. And FreeBSD we don't need to worry about, because `Mknod()` with a device number doesn't work in the first place. Devices can only be accessed through plain devfs. Let's use `int` then!